pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -468,17 +468,20 @@ public void redeliverUnacknowledgedMessages() {
     }
 
     @Override
-    public void redeliverUnacknowledgedMessages(Set<MessageIdImpl> messageIds) {
+    public void redeliverUnacknowledgedMessages(Set<MessageId> messageIds) {
```

Comment:
Thanks, As above reply, we also need this redeliverUnacknowledgedMessages method in Consumer.java  handling TopicMessageIdImpl. would like to change this in  Consumer.java  and make the case in each child class.

Reply:
In Consumer.java it doesn't know about the MessageIds. The call takes no parameters.

TopicsMessageIdImpl should be a specialization of MessageIdImpl (I assumed it was when I originally commented, until i saw Matteo's comment to the same effect). If it is a specialization of TopicsMessageIdImpl, then this signature doesn't need to change.

Reply:
Oh, it is in ConsumerBase.java, sorry for the wrong reference of Consumer.java. 
TopicsMessageIdImpl is more need to be a wrapper for MessageIdImpl, and as the reply below, may be better to implements MessageId, instead of extends MessageIdImpl.